src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -581,7 +582,7 @@ public virtual void Show(bool activate)
         public void SetParent(IWindowImpl parent)
         {
             _parent = (WindowImpl)parent;
-            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, _parent._hwnd);
+            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, _parent?._hwnd ?? IntPtr.Zero);
```

Comment:
We should probably set it to hidden window again if `_hiddenWindowIsParent == null`. Or introduce something like
```cs
private IntPtr EmptyParent => _hiddenWindowIsParent ? OffscreenParentWindow.Handle : IntPtr.Zero;
```
to avoid that check in multiple places.

Reply:
Good catch - updated that code. Can't use `_hiddenWindowIsParent` because that will only be set when the actual hidden window is the parent window, so need to look at `ShowInTaskbar` setting.